mutation_partition.cc

```diff
@@ -645,14 +645,15 @@ void write_cell(RowWriter& w, const query::partition_slice& slice, ::atomic_cell
 }
 
 template<typename RowWriter>
-void write_cell(RowWriter& w, const query::partition_slice& slice, const data_type& type, collection_mutation_view v) {
-    auto ctype = static_pointer_cast<const collection_type_impl>(type);
-    if (slice.options.contains<query::partition_slice::option::collections_as_maps>()) {
-        ctype = map_type_impl::get_instance(ctype->name_comparator(), ctype->value_comparator(), true);
+void write_cell(RowWriter& w, const query::partition_slice& slice, data_type type, collection_mutation_view v) {
```

Comment:
Why did you remove const& on |type|? data_type is a shared_ptr so it's cheaper to take it by reference when possible.

Reply:
Because of this:
`        type = map_type_impl::get_instance(ctype.name_comparator(), ctype.value_comparator(), true);
`

I could use `const abstract_type*` instead. Even though `get_instance` returns a `shared_ptr`, I could call `get()` on its result, because `type_interning_helper` keeps up the type's lifetime. That's an implementation detail though so I don't know if it's safe in the long run.